dune-project

```diff
@@ -28,51 +26,33 @@
     (>= 4.12.0)
     (< 5.0.0)))
   (batteries
-   (and
-    (>= 3.6.0)
-    (< 4.0)))
+   (>= 3.6.0))
```

Comment:
@smorimoto Thank you for citing the following page, I've read it:

> * Remove preventative upper bounds for dependencies
>   - https://github.com/ocaml/opam-repository/wiki/Policies#5-strict-dependency-constraints-should-be-avoided---or-preventive-upper-bounds

I think I understand the description of the policies on that page itself. Still, I couldn't find what will indeed happen when a _backward incompatible_ version of some dependency is released. Specifically, I need clarification on whether OPAM or I is responsible for resolving the incompatibility at that moment. (If it's not OPAM but me,  removing "preventive upper bounds" does not sound very reasonable to me.) Is that documented somewhere else?

Reply:
@gfngfn Basically, the responsibility lies with the maintainer of the opam repository as long as it's published in the opam repository. If not, it's generally considered the best way to cron GitHub Actions and regularly check that it's not broken. (So basically, repository owner, you're responsible.)
If this doesn't feel very reasonable, it probably makes sense to put upper bounds, but it's widely known that bulk dependency updates are more painful than gradual updates, so it's kind of a trade-off.

Reply:
Thank you for answering :pray: but I'm not very sure what the following means, unfortunately:

> it's widely known that bulk dependency updates are more painful than gradual updates

It's true that constraint solving can be more feasible without upper bounds than with upper bounds, but I don't think this implies that we shouldn't use upper bounds for backward compatibility; it will just incur the possibility of installing dependencies with interfaces that are unintended for users.

My original concern was that there will always be a possibility that a new backward-incompatible release of some dependencies breaks _already released versions_ of SATySFi that have no upper bounds. Indeed, breakdowns of this kind have happened a few times so far. (As far as I remember, backward-incompatible releases of `omd`, `camlimages`, and `core_kernel` have caused ones.)

Owing to this concern, I think it would be better to keep upper bounds, at least for our project. Does this make sense to you?

Reply:
We can consider it reasonable. Anyway, as the owner, you always have the right to decide everything!

Reply:
Reverted upper bound changes!

Reply:
Thank you very much for the update!

Btw, I'm a bit curious about what brought the OCaml community to the idea of avoiding upper bounds. It might be better for me to follow the original discussion in the community at some point.